roscpp_azure_iothub/src/ros_azure_iothub_cpp_node.cpp

```diff
@@ -413,13 +478,29 @@ static bool InitializeAzureIoTHub(ROS_Azure_IoT_Hub* iotHub)
     nh.getParam("connection_string", connectionString);
     ROS_INFO("connection_string: %s", connectionString.c_str());
 
+    std::string authenticationType;
+    nh.getParam("authentication_type", authenticationType);
+    ROS_INFO("authentication_type: %s", authenticationType.c_str());
+
     // Create the iothub handle here
     iotHub->deviceHandle = IoTHubDeviceClient_CreateFromConnectionString(connectionString.c_str(), protocol);
     if (iotHub->deviceHandle == NULL)
     {
         ROS_ERROR("Failure createing Iothub device.  Hint: Check you connection string.");
         return false;
     }
+
+    // Check if a x509 certificate should be used for authentication. If not,
+    // a sharedAccessKey is expected in the connection string.
+    if (authenticationType == g_authentication_x509)
```

Comment:
ROS_WARN if authentication type is unknown. Could be a mistype.

Reply:
I made a blank authentication type default to SAS. I then made it an error if the authentication type is unknown/invalid.